### PR TITLE
fix: side panel animation

### PR DESCRIPTION
**Pull request by Sumit4482**
[Video Preview](https://drive.google.com/file/d/1azGuWoUvG7yRG3HN-ldg4ggt5rd6E3Z3/view?usp=sharing) 

### Feature Preview

Fixes: Side Panel Animation
---

#### PR Checklist

- [✔️] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [✔️] I've listed at least one issue that this PR fixes in the description above.
- [✔️] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [✔️] All existing tests are passing.

### CODE REVIEW

File: frontend/appflowy_flutter/lib/workspace/presentation/widgets/edit_panel/panel_animation.dart [hunk 1]
```diff
@@ -30,12 +30,12 @@ class AnimatedPanelState extends State<AnimatedPanel> {
...
     final double duration = _isHidden && widget.isClosed ? 0 : widget.duration;
     return TweenAnimationBuilder(
-      curve: widget.curve ?? Curves.easeOut,
+      curve: widget.curve ?? Curves.easeInOut,
       tween: Tween<Offset>(
         begin: !widget.isClosed ? Offset.zero : closePos,
         end: !widget.isClosed ? Offset.zero : closePos,
       ),
-      duration: Duration(milliseconds: (duration * 1000).round()),
+      duration: Duration(milliseconds: (duration * 800).round()),
```
Comment (richardshiue) [inline frontend/appflowy_flutter/lib/workspace/presentation/widgets/edit_panel/panel_animation.dart:38]: Please revert or change to Duration(seconds: duration). This multiply by 1000 is supposed to convert from seconds to milliseconds.

File: frontend/.vscode/settings.json [hunk 2]
```diff
@@ -33,4 +33,5 @@
     },
     "editor.formatOnSave": true,
     "files.eol": "\n",
+    "cmake.configureOnOpen": false,
```
Comment (richardshiue) [inline frontend/.vscode/settings.json:36]: Only include changes relevant to the issue

**Review by richardshiue — COMMENTED** [review]
Hi @Sumit4482, thanks for the contribution. The sidebar isn't the only widget that is being animated when you hide or show the side panel. Could you take a look? If you look very closely, you can see from the video that the contents of the document don't animate in the same way that the sidebar does.

**Comment by Xazin** [comment]
I believe you only need to change a few lines of code in `appflowy_flutter\lib\workspace\presentation\home\desktop_home_screen.dart` - Between line and 264 and 288.

File: frontend/appflowy_flutter/lib/workspace/presentation/widgets/edit_panel/panel_animation.dart (hunk 1 above)
Comment (Sumit4482) [reply frontend/appflowy_flutter/lib/workspace/presentation/widgets/edit_panel/panel_animation.dart:38]: Okay got it !!

File: frontend/.vscode/settings.json (hunk 2 above)
Comment (Sumit4482) [reply frontend/.vscode/settings.json:36]: okay

**Comment by Sumit4482** [comment]
> I believe you only need to change a few lines of code in `appflowy_flutter\lib\workspace\presentation\home\desktop_home_screen.dart` - Between line and 264 and 288.

Yes, will do

**Review by richardshiue — CHANGES_REQUESTED** [review]
Hi @Sumit4482. Almost there! I have some more feedback.

- The animation curves should be consistent. I think we can use easeOutQuad for all
- The editPanel, homeStack, homeMenu and homeMenuResizer should all have this curve
- Please only include changes relevant to this PR. The vscode settings should be kept.

**Comment by richardshiue** [comment]
https://github.com/AppFlowy-IO/AppFlowy/issues/3315

**Review by richardshiue — COMMENTED** [review]
Much better, here's some more feedback. Also, please please please don't make changes to the .vscode jsons

**Comment by rileyhawk1417** [comment]
Well the animation side bar definitely feels snappy, even after spamming the shortcut a lot of times. Looking forward to see it merged 😁

**Comment by Sumit4482** [comment]
> Well the animation side bar definitely feels snappy, even after spamming the shortcut a lot of times. Looking forward to see it merged 😁


That's great to hear! 😄 Thanks for your feedback.

**Review by richardshiue — APPROVED** [review]
LGTM

**Pull request merged**